tpke/src/ciphertext.rs

```diff
@@ -25,13 +25,43 @@ impl<E: PairingEngine> Ciphertext<E> {
             (g_inv.clone(), E::G2Prepared::from(self.auth_tag)),
         ]) == E::Fqk::one()
     }
+
     fn construct_tag_hash(&self) -> E::G2Affine {
         let mut hash_input = Vec::<u8>::new();
         self.commitment.write(&mut hash_input).unwrap();
         hash_input.extend_from_slice(&self.ciphertext);
 
         hash_to_g2(&hash_input)
     }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        self.nonce.write(&mut bytes).unwrap();
+        bytes.extend_from_slice(&self.ciphertext);
+        self.auth_tag.write(&mut bytes).unwrap();
+        bytes
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        const NONCE_LEN: usize = 97;
```

Comment:
Shouldn't we make these constants as class constants somehow? I don't know how this works in Rust.
Also, is there any reference why is this 97 bytes? This is a G1 element, so in principle it should be possible to fit in 48 bytes in compressed form and 96 bytes in uncompressed. Same for the auth_tag element, which is from G2 (96 bytes in compressed form, 192 bytes in uncompressed)

Reply:
Can we used a compressed form for both G1 and G2 elements?

Reply:
In serialization, I will replace G1 and G2 usage with their compressed representation.